tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_opcodes.py

```diff
@@ -375,159 +406,1216 @@ def test_bal_extcodesize_and_oog(
 
 
 @pytest.mark.parametrize(
-    "fails_at_call", [True, False], ids=["oog_at_call", "successful_call"]
+    "oog_boundary",
+    [OutOfGasBoundary.SUCCESS, OutOfGasBoundary.OOG_BEFORE_TARGET_ACCESS],
+    ids=lambda x: x.value,
+)
+@pytest.mark.parametrize(
+    "target_is_warm", [False, True], ids=["cold_target", "warm_target"]
 )
-def test_bal_call_and_oog(
+@pytest.mark.parametrize(
+    "target_is_empty", [False, True], ids=["existing_target", "empty_target"]
+)
+@pytest.mark.parametrize("value", [0, 1], ids=["no_value", "with_value"])
+@pytest.mark.parametrize(
+    "memory_expansion", [False, True], ids=["no_memory", "with_memory"]
+)
+def test_bal_call_no_delegation_and_oog_before_target_access(
     pre: Alloc,
     blockchain_test: BlockchainTestFiller,
     fork: Fork,
-    fails_at_call: bool,
+    oog_boundary: OutOfGasBoundary,
+    target_is_warm: bool,
+    target_is_empty: bool,
+    value: int,
+    memory_expansion: bool,
 ) -> None:
-    """Ensure BAL handles CALL and OOG during CALL appropriately."""
+    """
+    CALL without 7702 delegation - test SUCCESS and OOG before target access.
+
+    When target_is_warm=True, we use EIP-2930 tx access list to warm the
+    target. Access list warming does NOT add to BAL - only EVM access does.
+    """
+    gas_costs = fork.gas_costs()
     alice = pre.fund_eoa()
-    bob = pre.fund_eoa()
+
+    target = (
+        pre.empty_account()
+        if target_is_empty
+        else pre.deploy_contract(code=Op.STOP)
+    )
+
+    ret_size = 32 if memory_expansion else 0
+
+    call_code = Op.CALL(
+        gas=0, address=target, value=value, ret_size=ret_size, ret_offset=0
```

Comment:
Side-note: this only target out_size for the memory. We also have in_size, this should not change anything but this (😅 ) should also be parametrized. (So this allocates memory for the return value of the call. Also have to allocate memory for the in-value of the call. If memory does not yet "exist" it will fill it with zeros (and also init calldata like that) but we still have to pay for this "expansion")